src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/TerminatedByNumberOfGenerations.java

```diff
@@ -0,0 +1,16 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import java.util.List;
+/**
+    termination criterion: number of generations
+ **/
+public class TerminatedByNumberOfGenerations implements TerminationCriterion{
+
+    @Override
+    public boolean readyToTerminate( final int generationNumber, final List<PhysicalPlanWithCost> currentGeneration, final List<List<PhysicalPlanWithCost>> previousGenerations ) {
+
+        return ( previousGenerations.size() >= generationNumber-1 );
```

Comment:
That's wrong. The threshold should be a member variable of this class (with a configurable value that is passed to the constructor). Then, in this method here, you have to check whether `generationNumber` exceeds the threshold.

Notice also that this termination criterion here completely ignores the other two arguments of this method (`currentGeneration` and `previousGenerations`).